### PR TITLE
fix: Select background error

### PR DESCRIPTION
**Pull request by DerrickTel**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Select background in disabled.  |
| 🇨🇳 Chinese | 修复 Select 组件 disabled 时背景颜色错误的问题. |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/select/style/index.less [hunk 1]
```diff
@@ -275,16 +275,20 @@
...
         }
       }
 
+      &-selected:not(&-disabled) {
+        background-color: @select-item-selected-bg;
+      }
+
       &-disabled {
         color: @disabled-color;
         cursor: not-allowed;
```
Comment (yoyo837) [inline components/select/style/index.less:294]: 直接给disabled加 个背景可以吗？

File: components/select/style/index.less [hunk 2]
```diff
@@ -285,6 +285,12 @@
         }
       }
 
+      &-selected {
+        &.@{select-prefix-cls}-item-option-disabled {
+          background-color: @input-disabled-bg;
+        }
+      }
+
       &-disabled {
```
Comment (yoyo837) [inline components/select/style/index.less:294]: 加到`&-disabled 下` 会不会更好？

Comment (DerrickTel) [reply components/select/style/index.less]: 😂谢谢大佬指导

**Comment by xrkffgg** [comment]
截个图 看下效果

**Comment by DerrickTel** [comment]
> 截个图 看下效果

image

**Comment by yoyo837** [comment]
可以错误的和修复后的都给个参考

**Comment by DerrickTel** [comment]
> 可以错误的和修复后的都给个参考

修改前
image
修改后
image

File: components/select/style/index.less [hunk 3]
```diff
@@ -286,6 +286,9 @@
       }
 
       &-disabled {
+        &.@{select-prefix-cls}-item-option-selected {
+          background-color: @input-disabled-bg;
```
Comment (xrkffgg) [inline components/select/style/index.less:290]: input 也不太合适

**Pull request merged**